appdatabase/database.go

```diff
@@ -76,6 +87,54 @@ func InitializeDB(path, password string, kdfIterationsNumber int) (*sql.DB, erro
 	return db, nil
 }
 
+func fixMissingKeyUIDForAccounts(sqlTx *sql.Tx) error {
```

Comment:
What you need to do here is:
- take the profile key uid from
  - the name of the folder where keystore files are stored or 
  - from the database name or
  - generate it since you have all you need to generate key uid for the derivation path `m`
- set the key uid from the previous point to all accounts which meet condition `where chat = 1 or wallet = 1 or type = generated`
- type `watch` is for watch only accounts and they should be without set key uid, so there is nothing to do here
- types `key` and `seed`, since there is no way to generate key uid for the derivation path `m` for such accounts, it's completely fine to leave them with an empty key uid, cause the effect should be the same as if you set any other key which is not referring to key uid of the master path. After a discussion with Andrea I figured out that the point of this is just for users to be able to move their funds to another account (not being stuck) even the UI may look broken or being unable to import a keypair (via seed phrase/private key) which was used in the past (old v1 app) for importing that account.

@qfrank please update this function according to what's said above and that's it.

Reply:
@qfrank the issue is in this line https://github.com/status-im/status-go/blob/85f9adf36263a0eb31b2a151ea2ee50d3c6331e8/api/old_mobile_user_upgrading_from_v1_to_v2_test.go#L95C2-L95C51
the keyuid which you're using for comparison is not the right one. The right one which should be set there is the one you get after `ImportMnemonic` call for certain seed phrase.

Reply:
This line is correct https://github.com/status-im/status-go/blob/85f9adf36263a0eb31b2a151ea2ee50d3c6331e8/api/old_mobile_user_upgrading_from_v1_to_v2_test.go#L104
but it's commented out.

Reply:
For accounts imported via private key is a different story, just because the address of an account being imported corresponds to the derivation path `m` and that's the reason why you got this line working https://github.com/status-im/status-go/blob/85f9adf36263a0eb31b2a151ea2ee50d3c6331e8/api/old_mobile_user_upgrading_from_v1_to_v2_test.go#L117C2-L117C50